pkg/sfu/forwarder.go

```diff
@@ -35,65 +35,38 @@ const (
 
 // -------------------------------------------------------------------
 
-type VideoStreamingChange int
+type VideoPauseReason int
 
 const (
-	VideoStreamingChangeNone VideoStreamingChange = iota
-	VideoStreamingChangePausing
-	VideoStreamingChangeResuming
+	VideoPauseReasonNone VideoPauseReason = iota
+	VideoPauseReasonMuted
+	VideoPauseReasonPubMuted
+	VideoPauseReasonFeedDry
+	VideoPauseReasonBandwidth
 )
 
-func (v VideoStreamingChange) String() string {
+func (v VideoPauseReason) String() string {
 	switch v {
-	case VideoStreamingChangeNone:
+	case VideoPauseReasonNone:
 		return "NONE"
-	case VideoStreamingChangePausing:
-		return "PAUSING"
-	case VideoStreamingChangeResuming:
-		return "RESUMING"
-	default:
-		return fmt.Sprintf("%d", int(v))
-	}
-}
-
-// -------------------------------------------------------------------
-
-type VideoAllocationState int
-
-const (
-	VideoAllocationStateNone VideoAllocationState = iota
-	VideoAllocationStateMuted
-	VideoAllocationStatePubMuted
-	VideoAllocationStateFeedDry
-	VideoAllocationStateAwaitingMeasurement
-	VideoAllocationStateOptimal
-	VideoAllocationStateDeficient
-)
-
-func (v VideoAllocationState) String() string {
-	switch v {
-	case VideoAllocationStateNone:
-		return "NONE"
-	case VideoAllocationStateMuted:
+	case VideoPauseReasonMuted:
 		return "MUTED"
-	case VideoAllocationStatePubMuted:
+	case VideoPauseReasonPubMuted:
 		return "PUB_MUTED"
-	case VideoAllocationStateFeedDry:
+	case VideoPauseReasonFeedDry:
 		return "FEED_DRY"
-	case VideoAllocationStateAwaitingMeasurement:
-		return "AWAITING_MEASUREMENT"
-	case VideoAllocationStateOptimal:
-		return "OPTIMAL"
-	case VideoAllocationStateDeficient:
-		return "DEFICIENT"
+	case VideoPauseReasonBandwidth:
+		return "BANDWIDTH"
 	default:
 		return fmt.Sprintf("%d", int(v))
 	}
 }
 
+// -------------------------------------------------------------------
+
 type VideoAllocation struct {
-	state              VideoAllocationState
-	change             VideoStreamingChange
+	pauseReason        VideoPauseReason
+	isDeficient        bool
```

Comment:
Changing this a bit to use a boolean to notify if track is deficient and a pause reason.

At some point want to get rid of availableLayers and exemptedLayers in forwarder fully (rely on bitrate measurement and opportunistic forwarding only), but it requires more thought.